lib/gems/pending/util/miq_object_storage.rb

```diff
@@ -15,4 +19,33 @@ def initialize(settings)
   def logger
     @logger ||= $log.nil? ? :: Logger.new(STDOUT) : $log
   end
+
+  private
+
+  DONE_READING = "".freeze
+  def read_single_chunk(chunksize = DEFAULT_CHUNKSIZE)
+    @buf_left ||= byte_count
+    return DONE_READING unless @buf_left.nil? || @buf_left.positive?
```

Comment:
`Excon` doesn't like `DONE_READING` being a frozen string since it has to do a `force_encoding` on it when it passes it back to the response handler.

We could remove the `.freeze`, but I think it just makes more sense to `.dup` the string here instead, so we are dealing with a mutated string when this is called multiple times in the same ruby process (I think most of our testing at the moment, it isn't).

Reply:
ok returning ```DONE_READING.dup``` instead.